python/tests/conftest.py

```diff
@@ -58,3 +72,66 @@ def resnet_model_uri(tmp_path_factory):
     model_uri = tmp_path / "resnet.pth"
     torch.save(resnet, model_uri)
     return model_uri
+
+
+@pytest.fixture
+def gcs_tmpdir() -> str:
+    """Create a temporary Google Cloud Storage (GCS) directory to test dataset.
+
+    To enable GCS test, it requires both the GCS credentials,
+    as well as `RIKAI_TEST_GCS_URL` being set.
+
+    Examples
+    --------
+
+    .. code-block:: bash
+
+        $ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/key.json
+        $ export RIKAI_TEST_GCS_URL=gs://bucket
+        $ pytest python/tests
+
+    References
+    ----------
+    https://cloud.google.com/dataproc/docs/concepts/connectors/cloud-storage
+    https://cloud.google.com/dataproc/docs/concepts/iam/iam
+    """
+
+    base_url = os.environ.get("RIKAI_TEST_GCS_URL", None)
+    if base_url is None:
+        pytest.skip("Skipping test. RIKAI_TEST_GCS_URL is not set")
```

Comment:
how would we distinguish between an actual error vs skipping here? e.g., if the env var is misconfigured?

Reply:
`pytest.skip` does not break the tests, while `raise ValueError` does.

The potential mistakely skipped case is that we offer a gs://url but does not have permission to read it. We can probably check the error message from the exception?

If we want to put hard enforcement on it, we can add an extra marker, like `--gcs-test`, and check it here.

```
if not pytest.markers.get("gcs_test"):
    return

assert <any error later>
```